axlearn/experiments/text/gpt/common.py

```diff
@@ -277,8 +269,7 @@ def model_config(
         layer_cfg.self_attention.attention.input_linear = attention_qkv_linear
     layer_cfg.self_attention.structure = atten_structure
     layer_cfg.self_attention.attention.atten_logit_cap = atten_logit_cap
-    if stack_cfg.klass is RepeatedTransformerLayer:
-        update_model_remat_config(stack_cfg=stack_cfg, layer_cfg=layer_cfg)
+    update_model_remat_config(stack_cfg=stack_cfg, layer_cfg=layer_cfg)
```

Comment:
Could you explain why we originally only did this for RepeatedTransformer and why it is okay to do it for everything now?

Reply:
I think it was used to avoid applying to pipelinedTransformer, where we may have explosion of activation memories due to saving intermediaries, and both stacked and repeated transformer are safe to apply the same remat thing here.

Reply:
We still have PipelinedTransformer in the codebase. Will this increase the memory usage of it?

Reply:
Technically, it is gated at a later stage, so it may be fine here. https://github.com/apple/axlearn/blob/main/axlearn/common/attention.py#L3994

Reply:
IIUC, you're saying the check this PR removes may never have been needed at all? Is this PR urgent? Ideally, it would be good to check with Mark.

Reply:
Or can we defensively leave the check for now with a TODO to revisit?

Reply:
How about adding
```suggestion
    if issubclass(stack_cfg.klass, (RepeatedTransformerLayer, StackedTransformerLayer)):
        update_model_remat_config(stack_cfg=stack_cfg, layer_cfg=layer_cfg)
```